### PR TITLE
Feature/grpc aws service scraper

### LINKED ISSUE #36: Add gPRC to aws-service and scrapers
when he request limit  is reached aws-service should renew scraper elastic ip to continue scraping

### PR DESCRIPTION
**Pull request by 1Jo1**
# Description

when he request limit is reached aws-service should renew scraper elastic ip to continue scraping

Fixes #36 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] test on a server(docker-compose)

# Checklist:

- [x] I have performed a self-review of my own code

### CODE REVIEW

File: aws_service/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM golang:1.13 as builder
+WORKDIR /go/src/github.com/codeuniversity/smag-mvp
```
Comment (alexmorten) [inline aws_service/Dockerfile]:
```suggestion
WORKDIR /app
```

File: aws_service/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+COPY . .
+RUN GO111MODULE=on CGO_ENABLED=0 GOOS=linux go build -installsuffix cgo -o aws_service awsService/main/main.go
```
Comment (alexmorten) [inline aws_service/Dockerfile]:
```suggestion
RUN CGO_ENABLED=0 GOOS=linux go build -installsuffix cgo -o aws_service awsService/main/main.go
```

File: aws_service/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+FROM alpine
+RUN apk --no-cache add ca-certificates
+RUN mkdir app
+COPY --from=builder /go/src/github.com/codeuniversity/smag-mvp/aws_service /app
```
Comment (alexmorten) [inline aws_service/Dockerfile]:
```suggestion
COPY --from=builder /app/aws_service /app
```

File: aws_service/proto/renewingAddress.proto [hunk 4]
```diff
@@ -0,0 +1,18 @@
+syntax = "proto3";
+
+package proto;
+
+service RouteGuide {
+    rpc renewElasticIp(RenewingElasticIp) returns (renewedElasticResult) {}
+}
+
+message renewedElasticResult {
```
Comment (alexmorten) [inline aws_service/proto/renewingAddress.proto]:
```suggestion
message RenewedElasticResult {
```

File: aws_service/renewing-address.go [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+)
+
+// Scraper represents the scraper containing all clients it uses
+type RenewingAddressGrpcServer struct {
+	awsSession *session.Session
+	ec2Service *ec2.EC2
+	grpcPort   int
+}
+
+// New returns an initilized scraper
```
Comment (jo-fr) [inline aws_service/renewing-address.go]:
```suggestion
// New returns an initialised gRPC Server
```
🙃

**Review by jo-fr — CHANGES_REQUESTED** [review]
please add docs

File: insta_comments-scraper/comments-scraper.go [hunk 6]
```diff
@@ -32,13 +32,13 @@ type PostCommentScraper struct {
...
 // New returns an initialized PostCommentScraper
-func New(postIDQReader *kafka.Reader, commentsInfoQWriter *kafka.Writer, errQWriter *kafka.Writer) *PostCommentScraper {
+func New(awsServiceAddress string, postIDQReader *kafka.Reader, commentsInfoQWriter *kafka.Writer, errQWriter *kafka.Writer) *PostCommentScraper {
 	s := &PostCommentScraper{}
 	s.postIDQReader = postIDQReader
 	s.commentsInfoQWriter = commentsInfoQWriter
 	s.errQWriter = errQWriter
 
-	s.httpClient = client.NewSimpleScraperClient()
+	s.httpClient = client.NewHttpClient(awsServiceAddress)
```
Comment (jo-fr) [inline insta_comments-scraper/comments-scraper.go]: please make configurable

**Review by jo-fr — CHANGES_REQUESTED** [review]

File: aws_service/renewing-address.go [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+}
+
+func (r *RenewingAddressGrpcServer) RenewElasticIp(context context.Context, reachedRequestLimit *pb.RenewingElasticIp) (*pb.RenewedElasticResult, error) {
+	ec2Address, err := r.getElasticPublicAddresses(reachedRequestLimit.InstanceId, reachedRequestLimit.PodIp)
+
+	log.Println("PodIp: ", reachedRequestLimit.PodIp)
+	log.Println("InstanceId: ", reachedRequestLimit.InstanceId)
+	if err != nil {
+		fmt.Println(err)
+		return &pb.RenewedElasticResult{IsRenewed: false}, nil
```
Comment (alexmorten) [inline aws_service/renewing-address.go]:
```suggestion
		return &pb.RenewedElasticResult{IsRenewed: false}, err
```

File: insta_posts-scraper/posts-scraper.go [hunk 8]
```diff
@@ -29,18 +29,17 @@ type InstaPostsScraper struct {
...
 // New returns an initilized scraper
-func New(nameQReader *kafka.Reader, infoQWriter *kafka.Writer, errQWriter *kafka.Writer) *InstaPostsScraper {
+func New(awsServiceAddress string, nameQReader *kafka.Reader, infoQWriter *kafka.Writer, errQWriter *kafka.Writer) *InstaPostsScraper {
 	i := &InstaPostsScraper{}
 	i.nameQReader = nameQReader
 	i.postsQWriter = infoQWriter
 	i.errQWriter = errQWriter
 
-	i.httpClient = scraper_client.NewSimpleScraperClient()
+	i.httpClient = scraper_client.NewHttpClient(awsServiceAddress)
```
Comment (alexmorten) [inline insta_posts-scraper/posts-scraper.go]: Can we make the usage of either `NewHttpClient` or `NewSimpleScraperClient` configurable? When we are not inside aws, we don't want to use `NewHttpClient `

File: scraper-client/http-client.go [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+
+	client.grpcClient, err = grpc.Dial(awsServiceAddress, grpc.WithInsecure())
+	if err != nil {
+		panic(err)
+	}
+	return client
+}
+
+func getAmazonInstanceId() (string, error) {
+	resp, err := http.Get("http://169.254.169.254/latest/meta-data/instance-id")
```
Comment (alexmorten) [inline scraper-client/http-client.go]: Should not be hard coded

**Review by alexmorten — CHANGES_REQUESTED** [review]

File: insta_posts-scraper/posts-scraper.go (hunk 8 above)
Comment (1Jo1) [reply insta_posts-scraper/posts-scraper.go]: yeah sure, @jo-fr  already suggested that

File: aws_service/renewing-address.go [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+				Values: aws.StringSlice([]string{instanceId}),
+			},
+			{
+				Name:   aws.String("private-ip-address"),
+				Values: aws.StringSlice([]string{localIp}),
+			},
+		},
+	})
+	if err != nil {
+		fmt.Println("Unable to elastic IP address: ", err)
```
Comment (alexmorten) [inline aws_service/renewing-address.go]: missing verb

File: aws_service/main/main.go [hunk 11]
```diff
@@ -0,0 +1,10 @@
+package main
+
+import (
+	"github.com/codeuniversity/smag-mvp/aws_service"
+)
+
+func main() {
+	s := aws_service.New(9900)
```
Comment (alexmorten) [inline aws_service/main/main.go]: The port should be configurable from outside :)

**Review by alexmorten — APPROVED** [review]
LGTM 👍 
Just two minor things

**Pull request merged**